pkg/controller/reconciler/deploymentresource_reconciler_test.go

```diff
@@ -0,0 +1,181 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crconfig "sigs.k8s.io/controller-runtime/pkg/config"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+const (
+	DeploymentResourceTestWaitDuration            = time.Second * 10
+	DeploymentResourceTestWaitInterval            = time.Second * 1
+	DeploymentResourceTestControllerDelayInterval = time.Millisecond * 100
+
+	TestDeploymentResourceNamespace           = "deploymentresource-basic"
+	TestDeploymentResourceName                = "test-deploymentresource"
+	TestDeploymentResourceRadiusResourceGroup = "default-deploymentresource-basic"
+)
+
+var (
+	TestDeploymentResourceScope = fmt.Sprintf("/planes/radius/local/resourcegroups/%s", TestDeploymentResourceRadiusResourceGroup)
+	TestDeploymentResourceID    = fmt.Sprintf("%s/providers/Microsoft.Resources/deployments/%s", TestDeploymentResourceScope, TestDeploymentResourceName)
+)
+
+func SetupDeploymentResourceTest(t *testing.T) (*mockRadiusClient, client.Client) {
+	SkipWithoutEnvironment(t)
+
+	// For debugging, you can set uncomment this to see logs from the controller. This will cause tests to fail
+	// because the logging will continue after the test completes.
+	//
+	// Add runtimelog "sigs.k8s.io/controller-runtime/pkg/log" to imports.
+	//
+	// runtimelog.SetLogger(ucplog.FromContextOrDiscard(testcontext.New(t)))
+
+	// Shut down the manager when the test exits.
+	ctx, cancel := testcontext.NewWithCancel(t)
+	t.Cleanup(cancel)
+
+	mgr, err := ctrl.NewManager(config, ctrl.Options{
+		Scheme: scheme,
+		Controller: crconfig.Controller{
+			SkipNameValidation: boolPtr(true),
+		},
+
+		// Suppress metrics in tests to avoid conflicts.
+		Metrics: server.Options{
+			BindAddress: "0",
+		},
+	})
+	require.NoError(t, err)
+
+	radius := NewMockRadiusClient()
+	err = (&DeploymentResourceReconciler{
+		Client:        mgr.GetClient(),
+		Scheme:        mgr.GetScheme(),
+		EventRecorder: mgr.GetEventRecorderFor("deploymentresource-controller"),
+		Radius:        radius,
+		DelayInterval: DeploymentResourceTestControllerDelayInterval,
+	}).SetupWithManager(mgr)
+	require.NoError(t, err)
+
+	go func() {
+		err := mgr.Start(ctx)
+		require.NoError(t, err)
+	}()
+
+	return radius, mgr.GetClient()
+}
+
+func Test_DeploymentResourceReconciler_Basic(t *testing.T) {
+	ctx := testcontext.New(t)
+	_, client := SetupDeploymentResourceTest(t)
+
+	name := types.NamespacedName{Namespace: TestDeploymentResourceNamespace, Name: TestDeploymentResourceName}
+	err := client.Create(ctx, &corev1.Namespace{ObjectMeta: ctrl.ObjectMeta{Name: name.Namespace}})
+	require.NoError(t, err)
+
+	deployment := makeDeploymentResource(name, TestDeploymentResourceID)
+	err = client.Create(ctx, deployment)
+	require.NoError(t, err)
+
+	// Deployment will update after operation completes
+	status := waitForDeploymentResourceStateReady(t, client, name)
+	require.Equal(t, TestDeploymentResourceID, status.Id)
+
+	err = client.Delete(ctx, deployment)
```

Comment:
We definitely need to test some of the novel delete behaviors. I'll be looking to see if that's tested as part of the DT controller.